src/game/client/line2d.cpp

```diff
@@ -0,0 +1,153 @@
+/**
+ * @file
+ *
+ * @author Jonathan Wilson
+ *
+ * @brief Line 2D
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "line2d.h"
+
+bool Clip_Line_2D(ICoord2D *p1, ICoord2D *p2, ICoord2D *c1, ICoord2D *c2, IRegion2D *clip_region)
+{
+    enum OutCodeEnum
+    {
+        CODE_INSIDE = 0, // 0000
+        CODE_LEFT = 1, // 0001
+        CODE_RIGHT = 2, // 0010
+        CODE_BOTTOM = 4, // 0100
+        CODE_TOP = 8, // 1000
+    };
+
+    int x_min = clip_region->lo.x;
+    int x_max = clip_region->hi.x;
+    int y_min = clip_region->lo.y;
+    int y_max = clip_region->hi.y;
+    int x0 = p1->x;
+    int y0 = p1->y;
+    int x1 = p2->x;
+    int y1 = p2->y;
+    int out_code0 = CODE_INSIDE;
+
+    if (p1->x < x_min) {
+        out_code0 = CODE_LEFT;
+    } else if (x0 > x_max) {
+        out_code0 = CODE_RIGHT;
+    }
+
+    if (y0 < y_min) {
+        out_code0 |= CODE_TOP;
+    } else if (y0 > y_max) {
+        out_code0 |= CODE_BOTTOM;
+    }
+
+    int out_code1 = CODE_INSIDE;
+
+    if (x1 < x_min) {
+        out_code1 = CODE_LEFT;
+    } else if (x1 > x_max) {
+        out_code1 = CODE_RIGHT;
+    }
+
+    if (y1 < y_min) {
+        out_code1 |= CODE_TOP;
+    } else if (y1 > y_max) {
+        out_code1 |= CODE_BOTTOM;
+    }
+
+    if (!(out_code1 | out_code0)) {
```

Comment:
At comment. `// Both points inside of region, no clipping required`

Reply:
fixed.